### ISSUE TITLE
AttackCapability of constructing/upgrading Buildings

### ISSUE THREAD

**Issue by KirmesBude** [stephenfournier/Dota-2-Building-Helper#69]
Buildings can already attack when they are still in construction.
Old Buildings can still attack while upgrading

**Repro Steps:**
Obvious

**Comment by MNoya** [comment]
Fix in builder.lua:

In `OnConstructionStarted`:

```lua
unit.original_attack = unit:GetAttackCapability()
`unit:SetAttackCapability(DOTA_UNIT_CAP_NO_ATTACK)`
```

In `OnConstructionCompleted`:

```lua
`unit:SetAttackCapability(unit.original_attack)`
```

**Comment by KirmesBude** [comment]
unit:SetAttackCapability(DOTA_UNIT_CAP_NO_ATTACK) does not stop the unit from attacking, so you would need to issue a stop command when implementing this for during-upgrades.

**Comment by MNoya** [comment]
> unit:SetAttackCapability(DOTA_UNIT_CAP_NO_ATTACK) does not stop the unit from attacking

What do you mean?

You'd have to change the attack capability of the building when the ability upgrade starts, that would be in StartUpgrade of buildings/upgrades.lua

**Comment by KirmesBude** [comment]
I ran into problems when the tower was currently attacking an enemy and I was upgrading it.
The tower kept attacking eventhough I set its AttackCapability to DOTA_UNIT_CAP_NO_ATTACK.
There was no projectile, but the enemy was still taking damage.

After manually issuing a stop command the tower stopped attacking (expected).
Trying to A-Click the enemy after that resulted in no attacking (hud errormsg and all)


It does not matter for OnConstructionStarted anyways. Just wanted to mention it.
